libraries/tools/kotlin-gradle-plugin/src/agp25/kotlin/org/jetbrains/kotlin/gradle/plugin/android/Android25ProjectHandler.kt

```diff
@@ -88,9 +88,15 @@ class Android25ProjectHandler(kotlinConfigurationTools: KotlinConfigurationTools
         else -> null
     }
 
-    override fun getJavaTask(variantData: BaseVariant): AbstractCompile? =
-            @Suppress("DEPRECATION") // There is always a Java compile task -- the deprecation was for Jack
-            variantData.javaCompile
+    override fun getJavaTask(variantData: BaseVariant): AbstractCompile? {
+        @Suppress("DEPRECATION") // There is always a Java compile task -- the deprecation was for Jack
+        return variantData::class.java.methods.firstOrNull { it.name == "getJavaCompileProvider" }
+            ?.invoke(variantData)
+            .let { taskProvider ->
```

Comment:
The control flow based on nullability seems to be wrong here, as it says: regardless of whether the `<...>?.invoke(...)` result is null or not, run the `let`
s `{ taskProvider -> ... }` block, possibly with `null` as `taskProvider`. Should it be `?.let { taskProvider -> ... }`?

Reply:
Thanks for spotting this. It's fixed now.